### DISCUSSION TITLE
v0.7.1

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#214] [Announcements]
1. 支持指定 Magpie 在特定的 GPU 上运行
2. 添加“在 3D 游戏中限制光标”的选项，开启后可以缓解 3D 游戏因独占光标而无法将光标限制在窗口内的问题
3. 修复在旧版本 Windows 上无法创建系统托盘图标的问题
4. 修复某些设备上无法初始化 Direct3D 的问题

<hr /><em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/v0.7.1'>v0.7.1</a>.</em>

**Comment by frdoma** [comment]
感谢发布新版，反馈两个在使用过程中发现的小问题：
1.标题栏的放大镜图标在白色背景下显示不全
![图标显示不全](https://user-images.githubusercontent.com/86701052/142732463-b70bffc1-72cb-48c3-9f47-48f550ac8dd3.png)
2.最小化到系统托盘后，右击任务栏的magpie图标退出软件图标依旧存在，把鼠标移动到图标上才会消失

**Reply by VladCorvi** [reply]
Please add NIS (NVIDIA Image Scaling) support!
请添加 NIS（NVIDIA 图像缩放）支持！

**Reply by hooke007** [reply]
> Please add NIS (NVIDIA Image Scaling) support! 请添加 NIS（NVIDIA 图像缩放）支持！

dup https://github.com/Blinue/Magpie/issues/213

**Reply by Blinue** [reply]
> 标题栏的放大镜图标在白色背景下显示不全

当系统为黑色主题时 Magpie 的图标会切换到白色，这时失去焦点的窗口标题栏也是白色的，因此会有这种情况。最根本的解决方法是重新设计图标

> 最小化到系统托盘后，右击任务栏的magpie图标退出软件图标依旧存在，把鼠标移动到图标上才会消失

我会修复这个 bug
master 分支已修复

**Reply by Blinue** [reply]
I will port NIS sooner or later

**Comment by Li4NwasTaken** [comment]
Thank you so much you are a godsent.

**Comment by urielejh** [comment]
Can this be of any usefulness? https://videocardz.com/newz/amd-releases-fidelityfx-super-resolution-plugin-for-unreal-engine-4

**Reply by Blinue** [reply]
With this plugin game developers can integrate FSR more easily, but it doesn't make sense for Magpie.

**Reply by urielejh** [reply]
yeah, i was thinking about any info/settings that could be integrated/ported to magpie : D

**Reply by Blinue** [reply]
"FSR Quality" only relates to the ratio between input and output resolution, so it's not applicable to Magpie. In addition, Magpie has ported CAS and you can modify ScaleModels.json to use it.

Thanks for the idea though.

**Comment by Coliostro** [comment]
DX10 card Ati Radeon 3850 using Magpie still doesn't work. Microsoft Basic Render Driver - Magpie works, performance is bad, down to 10-13 fps.   
<HTML>
<HEAD>
	<META HTTP-EQUIV="CONTENT-TYPE" CONTENT="text/html; charset=utf-8">
	<TITLE></TITLE>
	<META NAME="GENERATOR" CONTENT="OpenOffice 4.1.6  (Win32)">
	<STYLE TYPE="text/css">
	
	</STYLE>
</HEAD>
<BODY LINK="#000080" VLINK="#800000" DIR="LTR">
<TABLE WIDTH=100% BORDER=1 BORDERCOLOR="#000000" CELLPADDING=4 CELLSPACING=0>
	<COL WIDTH=128*>
	<COL WIDTH=128*>
	<TR VALIGN=TOP>
		<TD WIDTH=50%>
			<P><FONT SIZE=2>Description: ATI Radeon HD 3850<BR>2021-11-29
			18:16:12.962|info|Renderer. cpp:Renderer::_INITD3D|D3D Device has
			been created<BR>Functional level: 10.1<BR>2021-11-29
			18:16:12.963|info|Renderer. cpp:Renderer::_CREATESWAP
			CHAIN|Hardware Composition Support: <SPAN STYLE="background: #99ff66">No</SPAN><BR>2021-11-29
			18:16:12.963|info|Renderer. cpp:Renderer::_CREATESWAP
			Chain|Multiplier Overlay support: <SPAN STYLE="background: #99ff66">No</SPAN><BR>2021-11-29
			18:16:13.180|info|GraphicsCaptureFrameSource. Cpp:</FONT></P>
			<P><FONT SIZE=2>GRAPHICSCAPTUREFRAMESOURCE::Initialize|The current
			system does not have the IsBorderRequired API<BR>2021-11-29
			18:16:13.182|info|GraphicsCaptureFrameSource.
			cpp:gRaphicscApturefRamesOurces::Initialize|gRaphicscApturefRamesOurces
			initialization is complete<BR>2021-11-29 18:16:13.183|info|Utils.
			cpp:Utils::ReadFile|Read file:.
			\cache\FSR_EASU_hlsl_508533d5ddb3d6f30feec92f47e2a3a4c1821939.
			cmfx<BR>2021-11-29 18:16:13.183|info|EffectCache. cpp:
			Effectcache::Load|The cache has been read.
			\cache\FSR_EASU_hlsl_508533d5ddb3d6f30feec92f47e2a3a4c1821939.
			cmfx<BR>2021-11-29 18:16:13.183|info|EffectDrawer. cpp:
			EFFECTDRAWER::Initialize|Compile effects\FSR_EASU. hlsl takes
			0.801 milliseconds<BR>2021-11-29 18:16:13.183|info|Renderer.
			cpp:Renderer::GETSAMPLER|ID3D11SamplerState has been
			created<BR>Filter</FONT><FONT FACE="Lucida Sans Unicode"><FONT SIZE=2>：</FONT></FONT><FONT SIZE=2>D3D11_FILTER_MIN_MAG_MIP_POINT<BR>2021-11-29
			18:16:13.183|error|EffectDrawer.
			cpp:Effectdrawer::_pass::Initialize/Failed to create a pixel
			shader<BR>HRESULT</FONT><FONT FACE="Lucida Sans Unicode"><FONT SIZE=2>：</FONT></FONT><FONT SIZE=2>0x80070057<BR>2021-11-29
			18:16:13.184|error|EffectDrawer.
			cpp:Effectdrawer::Initialize|Pass1 failed to initialize<BR>2021-11-29
			18:16:13.184|error|Renderer.
			cpp:Renderer::INITIALIZEEFECTSANDCURSOR|_RESOLVEEFECTSJSON
			failed<BR>2021-11-29 18:16:13.184|critical|App.
			cpp:App::Run|Initialization effect failed, about to
			exit<BR>2021-11-29 18:16:13.187|info|App. cpp:App::_run|Start
			receiving window messages<BR>2021-11-29
			18:16:13.190|info|CursorDrawer.
			cpp:cursordrawer::~CURSORDRAWER|CURSORDRAWER has been
			destructed<BR>2021-11-29 18:16:13.196|info|App. cpp:App::_run|The
			main window has been destroyed<BR>2021-11-29
			18:16:13.196|info|DllMain. cpp:Run|App. Run failed<BR></FONT><BR>
			</P>
		</TD>
		<TD WIDTH=50%>
			<P><FONT SIZE=2>Description: Microsoft Basic Render
			Driver<BR>2021-11-29 18:20:06.229|info|Renderer.
			cpp:Renderer::_INITD3D|D3D Device has been created<BR>Functional
			level: 11.1<BR>2021-11-29 18:20:06.230|warning|Renderer.
			cpp:Renderer::_CREATESWAP CHAIN/Failed to obtain
			IDXGIOutput<BR>HRESULT</FONT><FONT FACE="Lucida Sans Unicode"><FONT SIZE=2>：</FONT></FONT><FONT SIZE=2>0x887A0004<BR>2021-11-29
			18:20:06.230|info|Renderer. cpp:Renderer::_CREATESWAP
			CHAIN|Hardware Composition Support: <SPAN STYLE="background: #99ff66">No</SPAN><BR>2021-11-29
			18:20:06.230|info|Renderer. cpp:Renderer::_CREATESWAP
			Chain|Multiplier Overlay support: <SPAN STYLE="background: #99ff66">No</SPAN><BR>2021-11-29
			18:20:06.448|info|GraphicsCaptureFrameSource. cpp:
			GRAPHICSCAPTUREFRAMESOURCE::Initialize|The current system does not
			have the IsBorderRequired API<BR>2021-11-29
			18:20:06.450|info|GraphicsCaptureFrameSource.
			cpp:gRaphicscApturefRamesOurces::Initialize|gRaphicscApturefRamesOurces
			initialization is complete<BR>2021-11-29
			18:20:06.450|info|EffectDrawer. cpp:
			EFFECTDRAWER::Initialize|Compile effects\FSR_EASU. hlsl takes
			0.362 milliseconds<BR>2021-11-29 18:20:06.450|info|Renderer.
			cpp:Renderer::GETSAMPLER|ID3D11SamplerState has been
			created<BR>Filter</FONT><FONT FACE="Lucida Sans Unicode"><FONT SIZE=2>：</FONT></FONT><FONT SIZE=2>D3D11_FILTER_MIN_MAG_MIP_POINT<BR>2021-11-29
			18:20:06.452|info|Utils. cpp:Utils::ReadFile|Read file:.
			\cache\FSR_RCAS_hlsl_4cb5627644735be449ae0fa7f3cd3404feb31290.
			cmfx<BR>2021-11-29 18:20:06.452|info|EffectCache. cpp:
			Effectcache::Load|The cache has been read.
			\cache\FSR_RCAS_hlsl_4cb5627644735be449ae0fa7f3cd3404feb31290.
			cmfx<BR>2021-11-29 18:20:06.452|info|EffectDrawer. cpp:
			EFFECTDRAWER::Initialize|Compile effects\FSR_RCAS. hlsl takes
			0.446 milliseconds<BR>2021-11-29 18:20:06.453|info|CursorDrawer.
			cpp:CURSORDRAWER::Initialize|The cursor movement speed has been
			adjusted<BR>2021-11-29 18:20:06.453|info|CursorDrawer.
			cpp:cursordrawer::Initialize|CURSORDRAWER initialization is
			complete<BR>2021-11-29 18:20:06.453|info|App. cpp:App::_run|Start
			receiving window messages<BR>2021-11-29
			18:20:21.844|info|Renderer. cpp:Renderer::_CHECKSRCSTATE|The
			foreground window has been changed<BR>2021-11-29
			18:20:21.844|info|Renderer. cpp:Renderer::_render|The status of
			the source window changes and exits full screen<BR>2021-11-29
			18:20:21.890|info|CursorDrawer.
			cpp:cursordrawer::~CURSORDRAWER|CURSORDRAWER has been
			destructed<BR>2021-11-29 18:20:21.944|info|App. cpp:App::_run|The
			main window has been destroyed<BR>2021-11-29
			18:20:21.944|info|DllMain. cpp:Run|About to exit</FONT></P>
		</TD>
	</TR>
</TABLE>
</BODY>
</HTML>

**Reply by Blinue** [reply]
Thanks for the feedback. Please delete the cache folder and then perform a scaling with "ATI Radeon HD 3850", the logs will be more useful then.

**Reply by Coliostro** [reply]
Thanks for the answer! I did as you said (fully fresh logs):
Runtime.txt:
2021-11-30 15:16:21.643|info|DllMain. cpp: setloglevel|Current log level: INFO
2021-11-30 15:16:21.643|info|App. cpp:App::Initialize|Initializing the App
2021-11-30 15:16:21.643|info|App. cpp: App::Initialize|WinRT has been initialized
2021-11-30 15:16:21.643|info|App. cpp: App::_REGISTERWNDCLASSES|Registered main window class
2021-11-30 15:16:21.643|info|App. cpp: App::_REGISTERWNDCLASSES|Registered DDF window class
2021-11-30 15:16:21.644|info|App. cpp:App::Initialize|App initialization was successful
2021-11-30 15:16:21.644|info|Utils. cpp:Utils::Hasher::Initialize|Utils::Hasher was successfully initialized
2021-11-30 15:16:57.509|info|DllMain. cpp:Run|OS version: 10.0.19042
2021-11-30 15:16:57.509|info|DllMain. cpp: Run|Source window title: Serious Sam HD: The Second Encounter
2021-11-30 15:16:57.509|info|App. cpp:App::Run/Runtime parameters：
hwndSrc：0x603bc
captureMode：0
adjustCursorSpeed：true
showFPS：true
disableRoundCorner：false
frameRate：-1
disableLowLatency：false
breakpointMode：false
disableWindowResizing：false
disableDirectFlip：false
ConfineCursorIn3DGames：false
adapterIdx：0
2021-11-30 15:16:57.509|info|App. cpp:App::Run|Source window customer area size: 1280x720
2021-11-30 15:16:57.522|info|App. cpp: App::_createhostwnd|Main window size: 1920x1080
2021-11-30 15:16:57.523|info|App. cpp: App::_createhostwnd|Main window has been created
2021-11-30 15:16:57.523|info|Renderer. cpp:Renderer::_INITD3D/Variable refresh rate support: Yes
2021-11-30 15:16:57.523|info|Renderer. cpp:LOGADAPTER|Current graphics adapter：
VID：0x1002
PID：0x9505
Description: ATI Radeon HD 3850
2021-11-30 15:16:57.577|info|Renderer. cpp:Renderer::_INITD3D|D3D Device has been created
Functional level: 10.1
2021-11-30 15:16:57.578|info|Renderer. cpp:Renderer::_CREATESWAP CHAIN|Hardware Composition Support: No
2021-11-30 15:16:57.578|info|Renderer. cpp:Renderer::_CREATESWAP Chain|Multiplier Overlay support: No
2021-11-30 15:16:57.796|info|GraphicsCaptureFrameSource. cpp: GRAPHICSCAPTUREFRAMESOURCE::Initialize|The current system does not have the IsBorderRequired API
2021-11-30 15:16:57.797|info|GraphicsCaptureFrameSource. cpp:gRaphicscApturefRamesOurces::Initialize|gRaphicscApturefRamesOurces initialization is complete
2021-11-30 15:16:58.034|info|EffectCache. cpp: Effectcache::Save|The cache has been saved. \cache\FSR_EASU_hlsl_508533d5ddb3d6f30feec92f47e2a3a4c1821939. cmfx
2021-11-30 15:16:58.034|info|EffectDrawer. cpp: EFFECTDRAWER::Initialize|Compile effects\FSR_EASU. hlsl takes 236.276 milliseconds
2021-11-30 15:16:58.034|info|Renderer. cpp:Renderer::GETSAMPLER|ID3D11SamplerState has been created
Filter：D3D11_FILTER_MIN_MAG_MIP_POINT
2021-11-30 15:16:58.034|error|EffectDrawer. cpp:Effectdrawer::_pass::Initialize/Failed to create a pixel shader
HRESULT：0x80070057
2021-11-30 15:16:58.034|error|EffectDrawer. cpp:Effectdrawer::Initialize|Pass1 failed to initialize
2021-11-30 15:16:58.034|error|Renderer. cpp:Renderer::INITIALIZEEFECTSANDCURSOR|_RESOLVEEFECTSJSON failed
2021-11-30 15:16:58.034|critical|App. cpp:App::Run|Initialization effect failed, about to exit
2021-11-30 15:16:58.037|info|App. cpp:App::_run|Start receiving window messages
2021-11-30 15:16:58.039|info|CursorDrawer. cpp:cursordrawer::~CURSORDRAWER|CURSORDRAWER has been destructed
2021-11-30 15:16:58.059|info|App. cpp:App::_run|The main window has been destroyed
2021-11-30 15:16:58.059|info|DllMain. cpp:Run|App. Run failed

Magpie.txt:
2021-11-30 15:16:21.1578|INFO|Magpie. App|Current log level: Info
2021-11-30 15:16:21.2198|INFO|Magpie. App/program launch
Process ID: 5096
Magpie version: 0.7.1.0
OS version: 10.0.19042
2021-11-30 15:16:21.2198|INFO|Magpie. App/Current language: ru-RU
2021-11-30 15:16:21.2368|INFO|Magpie. ScaleModelManager|Read scaling configuration
2021-11-30 15:16:21.3708|INFO|Magpie. ScaleModelManager|Monitoring ScaleModels. json changes
2021-11-30 15:16:21.5788|INFO|Magpie. MainWindow|Current hotkey: Alt+F11
2021-11-30 15:16:21.6358|INFO|Magpie. MagWindow|Creating a full-screen window in a new thread

**Reply by Blinue** [reply]
It's a bug, I've filed an issue for it #225

**Reply by Blinue** [reply]
[Magpie.zip](https://github.com/Blinue/Magpie/files/7634279/Magpie.zip)
Does this build work?

**Comment by okaxihd** [comment]
Hey, Blinue, are you going to add nvidia's NIS soon?

**Reply by hooke007** [reply]
https://github.com/Blinue/Magpie/issues/202

**Reply by Blinue** [reply]
NIS has been ported in the dev branch and will be released in v0.8

**Comment by MateoMatta** [comment]
Hey, Blinue, is it possible to add NVidia's DLSS in Magpie? (Despite the use of quick math with TensorCores...)
psdta: In love with Anime4K integration

**Reply by Blinue** [reply]
DLSS is not open source and may not be compatible with Magpie in license.

**Comment by MaxxximumPower** [comment]
Good morning

I have tested this software on some titles (Halo Infinity, Crsed TSR + FSR, Apex Legend, Battlefield 4, GTA V, FarCry 3, Hitman 2015, Quake Champion, F1 2018) it works on all 1600 * 900 to FHD.
PUBG runs for a few minutes then shuts down (but it could be the fault of the latest AMD divers 21.12.1

1) In the configuration of the ScaleModels.json file on the line I have configured this:

"effect": "FSR_RCAS",
 "sharpness": 0.001

However, I don't think there are any differences with the 0.1 setting (from gpuopen I had read somewhere that the configurations go from 0.1+). But I don't know the logic of this application.

RCAS is very "aggressive" and in the default value (0.87) it also modifies the colors. It seems that it cannot be zero: The program returns errors.

2) Also in Options -> Zoom -> Framerate

I activated Vsync (including panel activation) and when I hit 50 fp it is pretty fluid.
However at 40 fps it is not much, less than the native.
There is a rather annoying stutter.
If I turn off Vsync, the stuttering is continuous.

**Reply by hooke007** [reply]
the value of rcas in magpie: [0.00001,+∞)

**Reply by Blinue** [reply]
The larger the sharpness of rcas in magpie, the sharper the image

Some solutions to the stutters
1. change capture method
2. turn on "disable directflip" in settings
